File: output/leetcode/p24.cpp
// #include <iostream>
// using namespace std;
// struct ListNode {
//       int val;
//       ListNode *next;
//       ListNode() : val(0), next(nullptr) {}
//       ListNode(int x) : val(x), next(nullptr) {}
//       ListNode(int x, ListNode *next) : val(x), next(next) {}
//   };
// class Solution {
// public:
//     ListNode* swapPairs(ListNode* head) 
//     {
//         ListNode* dummynode =new ListNode(0);
//         dummynode->next =head;
//         ListNode* cur = dummynode;
//         while(cur->next!=nullptr&&cur->next->next!=nullptr)
//         {
//             ListNode* tmp= cur->next;
//             ListNode* tmp1 = cur->next->next->next;
//             cur->next = cur->next->next;   
//             cur->next->next = tmp;          
//             cur->next->next->next = tmp1;
//             cur=cur->next->next;
//         }
//         ListNode* result =dummynode -> next;
//           delete dummynode;
//         return result;
 

//     }
// };

File: output/python/p24.py

# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next

        

# class Solution:
#     def swapPairs(self,head):
#         dummyhead =ListNode(next=head)
#         current = dummyhead
#         while current.next and current.next.next:
#             temp=current.next
#             temp1=current.next.next.next
#             current.next = current.next.next
#             current.next.next = temp
#             temp.next = temp1
#             current = current.next.next
#         return dummyhead.next

    